compiler/src/iree/compiler/Codegen/Interfaces/PartitionableLoopsInterface.cpp

```diff
@@ -99,11 +102,12 @@ struct OuterParallelAsPartitionableLoops
       }
       partitionableLoops.push_back(iteratorType.index());
     }
-    if (partitionableLoops.size() > maxNumPartitionedLoops) {
-      partitionableLoops.erase(
-          partitionableLoops.begin(),
-          std::next(partitionableLoops.begin(),
-                    partitionableLoops.size() - maxNumPartitionedLoops));
+    if (maxNumPartitionedLoops.has_value() &&
+        partitionableLoops.size() > maxNumPartitionedLoops.value()) {
+      partitionableLoops.erase(partitionableLoops.begin(),
```

Comment:
Style nit. Just the indentations gotten a bit unreadable. Maybe flip the condition, so
```
if (!maxNumPartitionableLoops.has_value() || partitionableLoops.size() <= maxNumPartitionedLoops.value()) {
  return partitionableLoops;
}
partitionableLoops.erase(...)

```